packages/cmrjs/cmr.js

```diff
@@ -254,8 +114,16 @@ class CMR {
    * @returns {Promise.<Object>} the CMR response
    */
   async ingestGranule(xml) {
-    const headers = this.getHeaders(await this.getToken());
-    return _ingestConcept('granule', xml, 'Granule.GranuleUR', this.provider, headers);
+    let xmlString;
+    try {
+      xmlString = await readFile(xml, 'utf8');
```

Comment:
xml is a string, why do we read file here?  It may have caused the error:
`Error: ENAMETOOLONG: name too long, open '<?xml version="1.0" encoding="UTF-8" standalone="yes"?>`

Reply:
That is the current behavior in master: https://github.com/nasa/cumulus/blob/master/packages/cmrjs/cmr.js#L82

Reply:
Maybe checking the existence of the file, not trying to read it first.

Reply:
According to the [Node docs](https://nodejs.org/docs/v8.11.1/api/fs.html#fs_fs_exists_path_callback), "Using fs.exists() to check for the existence of a file before calling fs.open(), fs.readFile() or fs.writeFile() is not recommended. Doing so introduces a race condition, since other processes may change the file's state between the two calls. Instead, user code should open/read/write the file directly and handle the error raised if the file does not exist."

The problem here is that we're only checking for a `ENOENT` error code to determine if the file does not exist.  If we change line 122 to read something like `if (err.code === 'ENOENT' || err.code === 'ENAMETOOLONG') xmlString = xml;`, that should fix this problem.

Reply:
Added the catch error